code-studio/src/js/pd/workshop_dashboard/components/workshop_table.jsx

```diff
@@ -62,8 +67,9 @@ var WorkshopTable = React.createClass({
         url: '/api/v1/pd/workshops/' + workshop.id
       })
       .done(function () {
-        this.state.workshops.splice(workshop_index, 1);
-        this.setState(this.state);
+        var workshops = _.cloneDeep(this.state.workshops);
+        workshops.splice(workshop_index, 1);
+        this.setState({workshops: workshops});
```

Comment:
I think this could also be `this.setState({ workshops: [this.state.workshops[workshow_index]] })`?